admin/lint_json.m

```diff
@@ -0,0 +1,52 @@
+function lint_json(filesin, outputfile)
+    if isempty(filesin)
+        % Default to glob all
+        filesin = "**/*.m";
+    end
+    if isempty(outputfile) % Default to stdout
+        fh = 1;
+    else % Open file
+        fh = fopen(outputfile,'w');
+        if fh == -1
+            error("MATLAB:FileOpenError", "Failed to open file %s", outputfile);
+        end
+        cleanup = onCleanup(@()(fclose(fh)));
+    end
+
+
+    files = [];
+    for i = 1:numel(filesin)
+        flist = dir(filesin{i});
+        files = [files; arrayfun(@(file)(fullfile(file.folder, file.name)), flist, 'UniformOutput', false)];
+    end
+
+    issues = struct('issues', {{}}, 'size', 0);
+    raw = checkcode(files, '-id');
+    for i = 1:numel(raw)
+        for j = 1:numel(raw{i})
+            raw{i}(j).fileName = files{i};
+            curr = wng_compat(raw{i}(j));
+            issues.issues = {issues.issues{:}, curr};
+        end
+    end
+    issues.size = numel(issues.issues);
+    fprintf(fh, "%s", jsonencode(issues));
```

Comment:
`fprintf` has a limited buffer size, `fwrite` is probably the safer option here

Reply:
`fwrite` seems to be for writing raw data, we want formatted data. Unless I'm mistaken.
> `fwrite`
> Write data to binary file

https://uk.mathworks.com/help/matlab/ref/fwrite.html
https://uk.mathworks.com/help/matlab/ref/fprintf.html

Reply:
I only said this because I was recently bitten by using `fprintf` to write a large text file, and it stopped writing after ~255,000 characters. But that may have been the way I had called it (I had `fprintf(fh, long_char_array)` not `fprintf(fh, "%s", long_char_array)`.

Happy to leave this as is